### PR TITLE
test(checkout): CHECKOUT-7283 Introduce React Testing Library

### PR DESCRIPTION
**Pull request by bc-peng**
## What?
Introduce React Testing Library to `checkout-js`.

## Why?
The Enzyme library is no longer compatible with React 18.
As a result, we must switch to RTL to facilitate our migration to React 18.

## Testing / Proof
CI checks.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
       - run:
           name: "Run unit tests for core package"
           command: |
-            TEST=$(circleci tests glob "packages/core/**/*.spec.ts" "packages/core/**/*.spec.tsx" | circleci tests split --split-by=timings)
+            TEST=$(circleci tests glob "packages/core/**/*.spec.ts" "packages/core/**/*.spec.tsx" "packages/core/**/*.test.ts" "packages/core/**/*.test.tsx" | circleci tests split --split-by=timings)
```
Comment (bc-peng) [inline .circleci/config.yml:43]: Include `.test.ts` and `.test.tsx` in the `test-core` job.

File: packages/core/src/app/checkout/Checkout.test.tsx [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+        it('logs unhandled error', async () => {
+            checkout.use('CartWithShippingAndBilling');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForPaymentStep();
+
+            expect(defaultProps.errorLogger.log).toHaveBeenCalled();
+        });
```
Comment (bc-peng) [inline packages/core/src/app/checkout/Checkout.test.tsx:326]: Payment step initialization in local Node environment will result in the error shown below, thus no error mockup is required.
```
[InvalidArgumentError]: The provided URL must be absolute.
```

File: .eslintrc.json [hunk 3]
```diff
@@ -70,6 +70,13 @@
       "files": ["*.js", "*.jsx"],
       "extends": ["plugin:@nrwl/nx/javascript"],
       "rules": {}
+    },
+    {
+      "files": ["*.test.ts", "*.test.tsx"],
+      "extends": [
+        "plugin:testing-library/react",
+        "plugin:jest-dom/recommended"
+      ]
```
Comment (bc-peng) [inline .eslintrc.json:79]: I applied those esliint plugins in order to follow [the best practices](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-library-eslint-plugins) recommended by the creator of RTL.

File: packages/payment-integration-test-framework/src/react-testing-library-support/API.mock.ts
Comment (bc-peng) [inline packages/payment-integration-test-framework/src/react-testing-library-support/API.mock.ts:1]: All Ids and keys have been replaced with `x`.

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**